### PR TITLE
DRILL-7190: Missing backward compatibility for REST API with DRILL-6562

### PR DESCRIPTION
**Pull request by vdiravka**
See details:
https://issues.apache.org/jira/browse/DRILL-7190?focusedCommentId=16822315&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16822315

### CODE REVIEW

**Comment by vdiravka** [comment]
@sohami Please review

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 1]
```diff
@@ -221,6 +229,25 @@ private boolean isSupported(String format) {
             .collect(Collectors.toList());
   }
 
+  @POST
+  @Path("/storage/{name}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @Deprecated // use {@link #createOrUpdatePluginJSON} instead
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: use `@Deprecated(".....")`

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 2]
```diff
@@ -221,6 +229,25 @@ private boolean isSupported(String format) {
...
+  public JsonResult createOrUpdatePlugin(PluginConfigWrapper plugin) {
+    return createOrUpdatePluginJSON(plugin);
+  }
+
+  @GET
+  @Path("/storage/{name}/delete")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Deprecated // use the method with DELETE request {@link #deletePlugin(String)} instead
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: Same here.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 3]
```diff
@@ -147,9 +150,10 @@ public JsonResult enablePlugin(@PathParam("name") String name, @PathParam("val")
   }
 
   @GET
-  @Path("/storage/{name}/export/{format}")
+  @Path("/storage/{name}/export{format: (/[^/]+?)*}")
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: Please add Javadoc / comment explaining this logic.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 4]
```diff
@@ -147,9 +150,10 @@ public JsonResult enablePlugin(@PathParam("name") String name, @PathParam("val")
...
   @Produces(MediaType.APPLICATION_JSON)
   public Response exportPlugin(@PathParam("name") String name, @PathParam("format") String format) {
+    format = StringUtils.isNotEmpty(format) ? format.replace("/", "") : JSON_FORMAT;
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: Same here.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 5]
```diff
@@ -212,7 +216,11 @@ private boolean isSupported(String format) {
     return JSON_FORMAT.equalsIgnoreCase(format) || HOCON_FORMAT.equalsIgnoreCase(format);
   }
 
-  private List<PluginConfigWrapper> getConfigsFor(String pluginGroup) {
+  @GET
+  @Path("/storage{group: (/[^/]+?)*}-plugins.json")
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: Same here.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 6]
```diff
@@ -212,7 +216,11 @@ private boolean isSupported(String format) {
...
+  @Produces(MediaType.APPLICATION_JSON)
+  public List<PluginConfigWrapper> getConfigsFor(@PathParam("group") String pluginGroup) {
+    pluginGroup = StringUtils.isNotEmpty(pluginGroup) ? pluginGroup.replace("/", "") : ALL_PLUGINS;
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: Same here.

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java (hunk 1 above)
Comment (vdiravka) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java]: This annotation hasn't description parameter.
I have added javadoc description

**Comment by arina-ielchiieva** [comment]
Looks good, please squash the commits.

**Comment by vdiravka** [comment]
@arina-ielchiieva Done. Thanks for the review

**Comment by sohami** [comment]
@vdiravka and @arina-ielchiieva  - Thanks for the changes and review. Will merge it to master.

**Pull request merged**